### ISSUE TITLE
Marketing emails send to all customers, not just selected group

### ISSUE THREAD

**Issue by DaveRiba** [opencart/opencart#12645]
**What version of OpenCart are you reporting this for?**
4.0.2.2
**Describe the bug**
A clear and concise description of what the bug is
What section does it affect?
Sending an email to only Affiliates, sends to all customers
**To Reproduce**
1. Go to Marketing
2. Click on Mail
3. To: All Affiliates
4. Complete email
5. Send email
6. Every customer (including non-Affiliates) receives a copy of the email

**Expected behavior**
Email should only go to customers in the selected group
**Screenshots / Screen recordings**
If applicable, add screenshots or recordings to help explain your problem. Please keep these short and if you can, edit them to point out when/where the problem is.
https://www.useloom.com/ is perfect for screen recording with Chrome or with their desktop app

**Server / Test environment (please complete the following information):**
 - Local development? Deployed to a web server? Production
 - Operating system  Windows
 - PHP version  API20210902,NTS,VS16 
 - Apache version NA
 - Browser(s) tested with [e.g. chrome, safari - if applicable]
NA
**Additional context**
Thanks to ADD_Createive:
 looks like filter_affiliate has been removed from getCustomers() and the autocomplete. If that is intentional the the following bits of code would need updating to somehow get the affiliate's email addresses and use the affiliate autocomplete.

**Comment by mhcwebdesign** [comment]
More details please. Is it even a bug report at all? Or a feature request?

**Comment by DaveRiba** [comment]
No, it is a bug report
When you use the Mail feature to send emails, you have the option of recipient.
All Customers, all Newsletters, all Affiliates, etc
When you select a sub-group you would EXPECT OC would only send to that subgroup,
Not the case.  OC sends to ALL customers
That is a bug, not a feature.
I selected All Affiliates, and wanted emails to go to any customer listed as an Affiliate, but NO customers who were NOT affiliates.
OC sent to ALL CUSTOMERS

filter_affiliate has been removed from getCustomers() in OC 4
Needs to be reinstated
